target/linux/ath79/image/nand.mk

```diff
@@ -1,6 +1,7 @@
 define Device/glinet_gl-ar300m-nand
   ATH_SOC := qca9531
-  DEVICE_TITLE := GL-AR300M (NAND)
+  DEVICE_MODEL := GL-AR300M
+  DEVICE_VARIANT := NAND
```

Comment:
Is the intent that `DEVICE_VARIANT` describe the _board_ or the _firmware?_

In this case, `NAND` is describing a firmware variant that is available for the same board.

While I can definitely see something like this being of value for the GL-AR300M family:
* GL-AR300M -- with NAND
* GL-AR300M16 -- without NAND
* GL-AR300M-Lite -- without NAND and with only a single Ethernet port

it seems as though this is conflating the running firmware with the capabilities of the hardware

Reply:
Well, I would say the variant describes _anything_ that is used to distinguish devices beyond the name.

Reply:
@jeffsf If "GL-AR300M", "GL-AR300M16" and  "GL-AR300M-Lite" are official model names, then that should be used instead of the variant field.

The variant is for hardware revisions or other means of distinguishing the device, if no distinction by an _official_ model name is possible.

Reply:
Well, coming back to the original idea of this PR, the firmware variant could be a category of its own, like `DEVICE_FW_VARIANT := NAND`

One could then extend to:
`DEVICE_TITLE := $$(DEVICE_VENDOR) $$(DEVICE_MODEL)$$(if $$(DEVICE_VARIANT), $$(DEVICE_VARIANT),)$${if $$(value DEVICE_FW_VARIANT), ($$(DEVICE_FW_VARIANT)),}`
(Most probably I used the brackets in the wrong way, the intention is to enclose FW_VARIANT in normal () brackets here)

However, this is just an idea, I do not want to start discussion all over again.

Reply:
I'd be happy if VARIANT could hold *all* such special identifiers for now...

Reply:
I suggest to discuss this further in https://github.com/openwrt/openwrt/pull/2184 and keep it as it is for this PR.